python/triton/code_gen.py

```diff
@@ -1231,11 +1231,15 @@ def _init_kernel(self):
     def warmup(self, compile):
         return self._warmup(**compile, is_manual_warmup=True)
 
+    def _default_cache_dir(self):
```

Comment:
Do you prefer this to be a free function since it doesn't actually reference `self`?  Fine by me either way.

Reply:
Yeah, I think making it a free function will also make it easier for users to monkey-patch it if they want :D

Reply:
Sounds good, done!